libcudacxx/include/cuda/__floating_point/operators.h

```diff
@@ -0,0 +1,1250 @@
+      return __fp_ops::__le(__fp{__lhs}, __rhs);                                                                        \
+    }                                                                                                                   \
+    template <class _LhsConfig>                                                                                         \
+    _CCCL_NODISCARD _CCCL_HIDE_FROM_ABI _EXSPACE constexpr bool operator>(__fp<_LhsConfig> __lhs, _TYPE __rhs) noexcept \
+    {                                                                                                                   \
+      return __fp_ops::__gt(__lhs, __fp{__rhs});                                                                        \
+    }                                                                                                                   \
+    template <class _RhsConfig>                                                                                         \
+    _CCCL_NODISCARD _CCCL_HIDE_FROM_ABI _EXSPACE constexpr bool operator>(_TYPE __lhs, __fp<_RhsConfig> __rhs) noexcept \
+    {                                                                                                                   \
+      return __fp_ops::__gt(__fp{__lhs}, __rhs);                                                                        \
+    }                                                                                                                   \
+    template <class _LhsConfig>                                                                                         \
+    _CCCL_NODISCARD _CCCL_HIDE_FROM_ABI _EXSPACE constexpr bool operator>=(                                             \
+      __fp<_LhsConfig> __lhs, _TYPE __rhs) noexcept                                                                     \
+    {                                                                                                                   \
+      return __fp_ops::__ge(__lhs, __fp{__rhs});                                                                        \
+    }                                                                                                                   \
+    template <class _RhsConfig>                                                                                         \
+    _CCCL_NODISCARD _CCCL_HIDE_FROM_ABI _EXSPACE constexpr bool operator>=(                                             \
+      _TYPE __lhs, __fp<_RhsConfig> __rhs) noexcept                                                                     \
+    {                                                                                                                   \
+      return __fp_ops::__ge(__fp{__lhs}, __rhs);                                                                        \
+    }
+
+_LIBCUDACXX_FP_DEFINE_BINARY_OPERATORS_FOR(float, _CCCL_HOST_DEVICE)
```

Comment:
Instead of invoking this macro per each builtin type, it should be possible to make these operators also templated on the builtin type, and constrained on `cuda::std::is_floating_point`. That will significantly cut down on the sizes of overloads sets for them.